src/idl_gen_general.cpp

```diff
@@ -778,17 +778,29 @@ static void GenStruct(const LanguageParameters &lang, const Parser &parser,
       code += "}\n";
     }
     // Generate a ByteBuffer accessor for strings & vectors of scalars.
-    if (((field.value.type.base_type == BASE_TYPE_VECTOR &&
-          IsScalar(field.value.type.VectorType().base_type)) ||
-         field.value.type.base_type == BASE_TYPE_STRING) &&
-        lang.language == GeneratorOptions::kJava) {
-      code += "  public ByteBuffer ";
-      code += MakeCamel(field.name, lang.first_camel_upper);
-      code += "AsByteBuffer() { return __vector_as_bytebuffer(";
-      code += NumToString(field.value.offset) + ", ";
-      code += NumToString(field.value.type.base_type == BASE_TYPE_STRING ? 1 :
-                          InlineSize(field.value.type.VectorType()));
-      code += "); }\n";
+    if ((field.value.type.base_type == BASE_TYPE_VECTOR &&
+         IsScalar(field.value.type.VectorType().base_type)) ||
+         field.value.type.base_type == BASE_TYPE_STRING) {
+      switch (lang.language) {
+        case GeneratorOptions::kJava:
+          code += "  public ByteBuffer ";
+          code += MakeCamel(field.name, lang.first_camel_upper);
+          code += "AsByteBuffer() { return __vector_as_bytebuffer(";
+          code += NumToString(field.value.offset) + ", ";
+          code += NumToString(field.value.type.base_type == BASE_TYPE_STRING ? 1 :
+                              InlineSize(field.value.type.VectorType()));
+          code += "); }\n";
+          break;
+        case GeneratorOptions::kCSharp:
+          code += "  public ArraySegment<byte> Get";
+          code += MakeCamel(field.name, lang.first_camel_upper);
+          code += "Bytes() { var o = __offset(";
+          code += NumToString(field.value.offset);
+          code += "); return o != 0 ? __vector_as_arraysegment(o) : new ArraySegment<byte>(new byte[0], 0, 0); }\n";
```

Comment:
can you put this conditional inside `__vector_as_arraysegment` ? Also, rather than returning a blank segment, returning null would make more sense to me, which is also what the Java version does.